### PR TITLE
removed deprecated PauliTable, StabilizerTable and tests

### PR DESCRIPTION
**Pull request by adigaboy**
### Summary
Removed Deprecated code:
PauliTable
StabilizerTable
deprecated functions
and tests

### CODE REVIEW

**Comment by Raghav-Bell** [comment]
@adigaboy Please check lint tests locally (refer to [contribution guide](https://github.com/Qiskit/qiskit/blob/main/CONTRIBUTING.md)) and add release note. Also remove tests from [`test_pauli_sum_op.py`](https://github.com/Qiskit/qiskit/blob/main/test/python/opflow/test_pauli_sum_op.py#L42). As of now (latest), `0.22` deprecated code removal is requested (issue #10810 ). I don't think maintainers are going to remove deprecated code for `0.24` soon , let us wait for the  review from  maintainers.

**Comment by ikkoham** [comment]
@adigaboy Thank you very much. This is really important PR. Could you resolve the conflicts?

**Review by ikkoham — CHANGES_REQUESTED** [review]
This PR requires a release note for removal.

File: qiskit/quantum_info/__init__.py
Comment (ikkoham) [inline qiskit/quantum_info/__init__.py:36]: `pauli_basis` should not be removed.

**Review by ikkoham — DISMISSED** [review]
LGTM!

File: releasenotes/notes/remove-deprecated-in-quantum-info-0f9bd2b0c093307d.yaml
Comment (jakelishman) [inline releasenotes/notes/remove-deprecated-in-quantum-info-0f9bd2b0c093307d.yaml]: Could we just add a brief sentence here mentioning what the alternatives are (`PauliList` and `random_pauli_list`, I think?)?

Something like this, but feel free to use your own wording.
```suggestion
    The classes ``qiskit.quantum_info.PauliTable`` and `qiskit.quantum_info.StabilizerTable`
    are removed. The function ``random_pauli_table`` is also removed.
    They were deprecated in Qiskit 0.43 (with Terra 0.24), released in May 2023.
    Instead, you should use :class:`.PauliList` and :func:`.random_pauli_list`.
```

**Review by jakelishman — COMMENTED** [review]
Thanks for doing this, and thanks Hamamura-san for the reviewing and helping to fix up the last bits.  I just had one super minor comment about the release note, but otherwise let's get this merged.

File: releasenotes/notes/remove-deprecated-in-quantum-info-0f9bd2b0c093307d.yaml
Comment (jakelishman) [reply releasenotes/notes/remove-deprecated-in-quantum-info-0f9bd2b0c093307d.yaml]: (note that I had a couple of minor grammar tweaks in the suggestion as well as the extra sentence - I was expecting GitHub to highlight them in the diff, but it doesn't seem to have.)

Comment (ikkoham) [reply releasenotes/notes/remove-deprecated-in-quantum-info-0f9bd2b0c093307d.yaml]: Thank you very much. Yes, `PauliList` and `random_pauli_list` is the alternative.

**Review by ikkoham — APPROVED** [review]

**Pull request merged**
